pkg/admin.go

```diff
@@ -177,8 +190,12 @@ func (r *Registry) loadOrGenerateKey(identifier string) (map[string]interface{},
 	return crypto.JwkToMap(keyAsJwk)
 }
 
-func (r *Registry) signAndPublishEvent(eventType events.EventType, payload interface{}, signer func([]byte, time.Time) ([]byte, error)) (events.Event, error) {
-	event := events.CreateEvent(eventType, payload)
+func (r *Registry) signAndPublishEvent(eventType events.EventType, payload interface{}, previousEvent events.Event, signer func([]byte, time.Time) ([]byte, error)) (events.Event, error) {
+	var previousEventRef events.Ref
+	if previousEvent != nil {
+		previousEventRef = previousEvent.Ref()
```

Comment:
What if previous events don't have a ref?

Reply:
like old events

Reply:
Ref is always calculated on-the-spot, so that's not a problem. Although now I think of it, we still miss a check that it must match if it's included in the event.